src/execution.rs

```diff
@@ -474,6 +475,17 @@ impl Heap {
         }
     }
 
+    pub fn store_multiple(&mut self, start_address: u32, values: Vec<U256>) {
```

Comment:
```suggestion
    pub fn store_multiple(&mut self, start_address: u32, values: &[U256]) {
```
This avoids the need to clone vectors if the data needs to be reused later.

Reply:
Thats right, it would be more efficient if we had to reuse the vector. I'll make the change.
Anyways, I'm curious about this particular case. It shouldn't be any performance difference if we don't want to use the vector after calling this function, right? Using a reference vs moving the value to the callee should be the same.

Reply:
It should be about the same, yes. The memcpy/load used by the move of `Vec` is slightly larger than it is for the slice (3 vs 2 machine words), but I would be surprised if it becomes relevant except in very edge cases.

Reply:
Great to know, thanks!